janus_core/helpers/janus_types.py

```diff
@@ -179,3 +131,122 @@ class EoSResults(TypedDict, total=False):
     bulk_modulus: float
     v_0: float
     e_0: float
+
+
+# pylint: disable=too-few-public-methods
+class Observable:
```

Comment:
This is probably complex enough now that it should be in its own module, or a module related to correlations/observables.

Reply:
I think it is yes, but I was hesitant. What do we call it? Since it will then form part of a user facing api t might be confusing to have ```janus_core.helpers.observables``` and ```janus_core.helpers.observable``` and the like.

Reply:
Any reason not to just put it in `janus_core.helpers.observables` itself?

Reply:
> Any reason not to just put it in `janus_core.helpers.observables` itself?

I seem to recall some circular import issues, I will check

Reply:
The issue is I want to use ```StartStopStep``` and ```SliceLike``` in ```observables.py```, imported from ```janus_types.py```.

But also use ```Observable``` in ```janus_types.py``` imported from ```observable.py``` for ```CorrelationKwargs```, its a circle.

```
ImportError: cannot import name 'SliceLike' from partially initialized module 'janus_core.helpers.janus_types' (most likely due to a circular import) (/home/vol04/scarf1214/janus-core_devel/janus_core/helpers/janus_types.py)
```

It can be solved by double-defining ```StartStopStep``` and ```SliceLike``` in ```observables.py```... or maybe there is a better python solution I do not know?

```CorrelationKwargs``` is the only ```janus_type``` that requires another janus object in its definition. This is since ```Observable``` has increased in complexity to support the atom and dimension averaging automatically.

Reply:
I suppose one option could be to also move `CorrelationKwargs`, but I think that ought to be in `janus_types.py`, so it's not a particularly nice solution.

Assuming `SliceLike` is only used as a type hint, what about about gating the import with `TYPE_CHECKING`, or does it throw errors?

Reply:
Ok nice, that's probably the best option.

You can also add `from __future__ import annotations`, which means you can get rid of the quotes (ruff will then probably also ask you to convert `Union` to `x | y` etc.)